data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisClientApiHandler.java

```diff
@@ -30,44 +33,84 @@ public class KinesisClientApiHandler {
 
     private final Backoff backoff;
     private final KinesisAsyncClient kinesisClient;
-    private int failedAttemptCount;
     private int maxRetryCount;
 
     public KinesisClientApiHandler(final KinesisAsyncClient kinesisClient, final Backoff backoff, final int maxRetryCount) {
         this.kinesisClient = kinesisClient;
         this.backoff = backoff;
-        this.failedAttemptCount = 0;
         if (maxRetryCount <= 0) {
             throw new IllegalArgumentException("Maximum Retry count should be strictly greater than zero.");
         }
         this.maxRetryCount = maxRetryCount;
     }
 
+    public StreamIdentifier getStreamIdentifierFromStreamArn(final String streamArnString) {
```

Comment:
I think you can make this package protected. Remove the `public` modifier.

Reply:
Based on my other comments, I think this code can be refactored to have a single place for handling the stream identification.